src/validateAtoms.ts

```diff
@@ -0,0 +1,72 @@
+import { atom, SetStateAction, WritableAtom } from 'jotai';
+import { loadable } from 'jotai/utils';
+
+import type { CommonState } from './atomWithValidate';
+
+export type GetValues = <Value>(labeledAtoms: LabeledAtoms<Value>) => {
+  [k: string]: Value;
+};
+export type Validator = <Value>(values: { [k: string]: Value }) => Promise<any>;
```

Comment:
Maybe something like this?
```suggestion
export type Validator = <Values extends Record<string, unknown>>(values: Values) => Promise<any>;
```